sdks/java/extensions/sql/zetasql/src/main/java/org/apache/beam/sdk/extensions/sql/zetasql/translation/SqlOperators.java

```diff
@@ -80,58 +81,59 @@
+      createBuiltinFunctionOperator("CONCAT", BeamBuiltinMethods.CONCAT_METHOD);
 
   public static final SqlOperator REPLACE =
-      createUdfOperator("REPLACE", BeamBuiltinMethods.REPLACE_METHOD);
+      createBuiltinFunctionOperator("REPLACE", BeamBuiltinMethods.REPLACE_METHOD);
 
-  public static final SqlOperator TRIM = createUdfOperator("TRIM", BeamBuiltinMethods.TRIM_METHOD);
+  public static final SqlOperator TRIM =
+      createBuiltinFunctionOperator("TRIM", BeamBuiltinMethods.TRIM_METHOD);
 
   public static final SqlOperator LTRIM =
-      createUdfOperator("LTRIM", BeamBuiltinMethods.LTRIM_METHOD);
+      createBuiltinFunctionOperator("LTRIM", BeamBuiltinMethods.LTRIM_METHOD);
 
   public static final SqlOperator RTRIM =
-      createUdfOperator("RTRIM", BeamBuiltinMethods.RTRIM_METHOD);
+      createBuiltinFunctionOperator("RTRIM", BeamBuiltinMethods.RTRIM_METHOD);
 
   public static final SqlOperator SUBSTR =
-      createUdfOperator("SUBSTR", BeamBuiltinMethods.SUBSTR_METHOD);
+      createBuiltinFunctionOperator("SUBSTR", BeamBuiltinMethods.SUBSTR_METHOD);
 
   public static final SqlOperator REVERSE =
-      createUdfOperator("REVERSE", BeamBuiltinMethods.REVERSE_METHOD);
+      createBuiltinFunctionOperator("REVERSE", BeamBuiltinMethods.REVERSE_METHOD);
 
   public static final SqlOperator CHAR_LENGTH =
-      createUdfOperator("CHAR_LENGTH", BeamBuiltinMethods.CHAR_LENGTH_METHOD);
+      createBuiltinFunctionOperator("CHAR_LENGTH", BeamBuiltinMethods.CHAR_LENGTH_METHOD);
 
   public static final SqlOperator ENDS_WITH =
-      createUdfOperator("ENDS_WITH", BeamBuiltinMethods.ENDS_WITH_METHOD);
+      createBuiltinFunctionOperator("ENDS_WITH", BeamBuiltinMethods.ENDS_WITH_METHOD);
 
   public static final SqlOperator LIKE =
-      createUdfOperator("LIKE", BeamBuiltinMethods.LIKE_METHOD, SqlSyntax.BINARY);
+      createBuiltinFunctionOperator("LIKE", BeamBuiltinMethods.LIKE_METHOD, SqlSyntax.BINARY);
 
   public static final SqlOperator VALIDATE_TIMESTAMP =
-      createUdfOperator(
+      createBuiltinFunctionOperator(
           "validateTimestamp",
           DateTimeUtils.class,
           "validateTimestamp",
           x -> NULLABLE_TIMESTAMP,
           ImmutableList.of(TIMESTAMP));
 
   public static final SqlOperator VALIDATE_TIME_INTERVAL =
-      createUdfOperator(
+      createBuiltinFunctionOperator(
           "validateIntervalArgument",
           DateTimeUtils.class,
           "validateTimeInterval",
           x -> NULLABLE_BIGINT,
           ImmutableList.of(BIGINT, OTHER));
 
   public static final SqlOperator TIMESTAMP_OP =
-      createUdfOperator("TIMESTAMP", BeamBuiltinMethods.TIMESTAMP_METHOD);
+      createBuiltinFunctionOperator("TIMESTAMP", BeamBuiltinMethods.TIMESTAMP_METHOD);
 
   public static final SqlOperator DATE_OP =
-      createUdfOperator("DATE", BeamBuiltinMethods.DATE_METHOD);
+      createBuiltinFunctionOperator("DATE", BeamBuiltinMethods.DATE_METHOD);
```

Comment:
Changes like this raise flags. If you haven't watched [how to grow a language](https://www.youtube.com/watch?v=_ahvzDzKdB0) I highly recommend it. TL;DR built-in and user-defined functions should have little to no difference if it can be avoided. (for zetasql I think the distinction is "do we execute in Java or do we execute via gRPC to C++"

Reply:
My understanding is `createUdfOperator` was to create functions execute via gRPC to C++. The naming was confusing and this is a change to how what "DATE" should be executed.

Reply:
I only added `createBuiltinFunctionOperator` to so we could have a default function group. I could remove it and always require a function group.

This is another thing that is liable to change if we steer away from `Method`.

Reply:
I removed `createBuiltinFunctionOperator` and required all function definitions to specify a function group.